ui/src/components/app-form/app-form.tsx

```diff
@@ -433,321 +624,360 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
+              <Controller
+                name="branch"
+                control={control}
+                render={({ field }) => (
+                  <FormControl>
+                    <TextField
+                      {...field}
+                      id="branch"
+                      label="Branch"
+                      placeholder="e.g., main"
+                      defaultValue={formData?.defaultBranch || 'main'} // Prepopulate if data is fetched
+                      disabled={!isUrlValid} // Disable field until URL is validated
+                    />
+                  </FormControl>
+                )}
               />
-            </FormControl>
-          )}
-        />
-
-        <Controller
-          name="description"
-          control={control}
-          render={({ field }) => (
-            <FormControl
-              fullWidth
-              className="form-control outer-div"
-              variant="outlined"
-              style={{ position: 'relative' }}
-            >
-              <label
-                htmlFor="description"
-                className="description-label"
-                style={{
-                  position: 'absolute',
-                  top: '0',
-                  left: '10px',
-                  transform: 'translate(0, -50%)',
-                  color: isFocused ? '#ba18da' : '#646464',
-                  backgroundColor: '#fafafa',
-                  padding: '0 4px',
-                  zIndex: 1,
-                }}
-              >
-                Description
-              </label>
-              <textarea
-                {...field}
-                ref={textAreaRef}
-                id="description"
-                value={description}
-                onChange={(e) => {
-                  setDescription(e.target.value);
-                  field.onChange(e.target.value);
-                  adjustTextareaHeight(e.target);
-                }}
-                onFocus={() => handleFocus(true)}
-                onBlur={() => {
-                  field.onBlur();
-                  handleFocus(false);
-                }}
-                className="description_text-field"
-                placeholder="Add app description (max. 200 characters)"
-                style={{
-                  paddingBottom: '8px',
-                  borderColor: isFocused ? '#ba18da' : '#ccc',
-                  letterSpacing: '.15px',
-                  backgroundColor: '#FAFAFA',
-                }}
+              {/* Conda YAML Directory Input */}
+              <Controller
+                name="conda_project_yml"
+                control={control}
+                render={({ field }) => (
+                  <FormControl fullWidth>
+                    {/* Label for Conda YAML Directory */}
+                    <InputLabel
+                      htmlFor="conda_project_yml"
+                      shrink
+                      sx={labelStyle}
+                    >
+                      Conda YAML Directory
```

Comment:
As Conda YAML could mean multiple things, like conda environment yaml.

```suggestion
                      Conda Project YAML Directory
```

Reply:
Updated the label